assets-and-images.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: page
+title: Assets and Images
+permalink: /assets-and-images/
+---
+
+Flutter applications include both code and assets (sometimes called resources).
+Assets can include static data, configuration files, or anything else an
+application needs to function; in particular, assets include icons and other
+images displayed in the UI.
+
+## Specifying assets
+
+The flutter build process uses the `flutter.yaml` file to identify assets
+required by an application, as shown in the [Tutorial](/tutorial/). There are
+two sections that specify assets: the general `assets` section and the
+`material-design-icons` section, which includes any
+[Material Design icons](https://design.google.com/icons/) the application uses.
+Both kinds of assets are included in the Flutter application's _asset bundle_,
+which is a special archive that Flutter applications can read from at runtime.
+
+### Asset variants
+
+The build process supports the notion of asset variants: different versions of
+an asset that might be displayed in different contexts. When an asset's path is
+specified in the `assets` section of `flutter.yaml`, the build process looks for
+any files with the same name in subdirectories next to the main asset. Such
+files are then included in the asset bundle along with the main asset.
+
+The key use of asset variants is in choosing resolution appropriate images; see
+below. In the future, this mechanism may be extended to include variants for
+different locales or regions, reading directions, etc.
+
+## Working with assets
+
+Your application can access its assets through the
+[AssetBundle](http://docs.flutter.io/flutter/services/AssetBundle-class.html)
+interface. The two main methods on an asset bundle allow you to load a string
+or an image out of the bundle, given a logical key.  The logical key maps to
+the path to the asset specified in the `flutter.yaml` file at build time. The
+application's asset bundle is accessible through the
+[DefaultAssetBundle](http://docs.flutter.io/flutter/widgets/DefaultAssetBundle-class.html)
+widget.
+
+As a shortcut, the
+[AssetImage](http://docs.flutter.io/flutter/widgets/AssetImage-class.html)
+widget provides a simple way to display an image in the UI.  Images may also be
+specified as the background of a
```

Comment:
This sentence seems like it "came out of nowhere". Consider removing, or going into more detail why it's relevant/connected to the point of this paragraph.

(I might be missing something)